streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -409,6 +411,10 @@ public void close() {
                 } catch (final Throwable e) {
                     log.error("Failed to close producer due to the following error:", e);
                 }
+            } else {
+                for (final StreamsProducer streamsProducer : taskProducers.values()) {
+                    streamsProducer.close();
+                }
```

Comment:
This issues was introduced in the PR that introduced `StreamsProducer` -- we forgot to close them. Fixing this on the side.